qt-bugs@ issue : N216195
Trolltech task ID : 216537 (Applied in Qt 4.5)
bugs.kde.org number : 162737
applied: no
author: Michael Leupold <lemma@confuego.org>

This patch fixes some characters in a tabbar being cut-off due
to a wrong size hint (visible if using a kstyle).

--- src/gui/widgets/qtabbar.cpp	(Revision 820510)
+++ src/gui/widgets/qtabbar.cpp	(Arbeitskopie)
@@ -1084,7 +1084,7 @@
         QStyleOptionTabV2 opt;
         initStyleOption(&opt, index);
         opt.text = d->tabList.at(index).text;
-        QSize iconSize = tab->icon.isNull() ? QSize() : opt.iconSize;
+        QSize iconSize = tab->icon.isNull() ? QSize(0, 0) : opt.iconSize;
         int hframe  = style()->pixelMetric(QStyle::PM_TabBarTabHSpace, &opt, this);
         int vframe  = style()->pixelMetric(QStyle::PM_TabBarTabVSpace, &opt, this);
         const QFontMetrics fm = fontMetrics();
